pilot/pkg/model/context.go

```diff
@@ -40,16 +41,14 @@ type Environment struct {
 	// Mesh is the mesh config (to be merged into the config store)
 	Mesh *meshconfig.MeshConfig
 
-	// Mixer subject alternate name for mutual TLS
-	MixerSAN []string
-
-	// PushContext holds informations during push generation. It is reset on config change, at the beginning
-	// of the pushAll. It will hold all errors and stats and possibly caches needed during the entire cache computation.
+	// Mutex is used to protect PushContext
+	Mutex sync.RWMutex
+	// PushContext holds information during push generation. It is reset on config change, at the beginning
+	// of the full Push. It will hold all errors and stats and possibly caches needed during the entire cache computation.
 	// DO NOT USE EXCEPT FOR TESTS AND HANDLING OF NEW CONNECTIONS.
 	// ALL USE DURING A PUSH SHOULD USE THE ONE CREATED AT THE
 	// START OF THE PUSH, THE GLOBAL ONE MAY CHANGE AND REFLECT A DIFFERENT
 	// CONFIG AND PUSH
-	// Deprecated - a local config for ads will be used instead
```

Comment:
Can you describe the motivation a bit more here? Is seems like you're taking a step backward from this deprecation comment.

@costinm @rshriram @howardjohn  thoughts?

Reply:
This comment is not right,  pushContext is widely used during  push.

Reply:
I don't think that's right, for init discovery response we use global PushContext but for new ones we generate a new context and pass that through?